examples/c/alexnet.c

```diff
@@ -5,11 +5,11 @@
 typedef shape_t const *p_shape_t;
 
 // https://papers.nips.cc/paper/4824-imagenet-classification-with-deep-convolutional-neural-networks.pdf
-s_model_t *alexnet(shape_t *image_shape, uint32_t arity)
+s_model_t *alexnet(const shape_t *image_shape, uint32_t arity)
 {
-    shape_ctx_t *sc = make_shape_ctx();
-    trait_ctx_t *tc = make_trait_ctx();
-    s_model_ctx_t *ctx = new_s_model_ctx();
+    shape_ctx_t *sc = new_shape_ctx();
+    trait_ctx_t *tc = new_trait_ctx();
+    s_model_ctx_t *ctx = make_s_model_ctx();
```

Comment:
why shape and trait context are **new** and s_model context is **made**?

Reply:
pointers obtained from `make_` functions are GCed automatically,
pointers obtained from `new_` functions needs to be freed via `del_` manually.